atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventLog.java

```diff
@@ -120,6 +121,13 @@ private IdentifiedVersion getLatestVersionAndVerify(IdentifiedVersion endVersion
 
     private void processSuccess(LockWatchStateUpdate.Success success) {
         Preconditions.checkState(latestVersion.isPresent(), "Must have a known version to process successful updates");
+        Optional<IdentifiedVersion> snapshotVersion = snapshot.getSnapshotVersion();
+        Preconditions.checkState(snapshotVersion.isPresent(),
+                "Must have a snapshot before processing successful updates");
```

Comment:
Already enforced elsewhere (i.e. must have a snapshot before processing successes), but this is just because we have to handle optionals everywhere.